python/ml4ir/base/model/architectures/auto_dag_network.py

```diff
@@ -274,23 +318,33 @@ def instantiate_op(self, current_layer_type: str, current_layer_args: Dict) -> l
             raise KeyError(f"Layer type: '{current_layer_type}' "
                            f"is not supported or not found in subclasses of keras.layers.Layer")
 
-    def get_layer_op(self, layer_args: Dict) -> Dict[str, Union[str, bool, layers.Layer]]:
+    def get_layer_op(
+            self,
+            layer_args: Dict,
+            existing_op: layers.Layer = None
+    ) -> Dict[str, Union[str, bool, layers.Layer]]:
         """
         Define the layer operation for a layer in the model config
 
         Parameters
         ----------
         layer_args: dict
             All the arguments from model config which are needed to define a layer op
+        existing_op: instance of type keras.layers.Layer
+            If specified, reuse the layer op
 
         Returns
         -------
         dict
             Dictionary with the layer instance, inputs required for the layer and the format of inputs to the layer
         """
+        if existing_op and existing_op.__class__.__name__ != layer_args[self.LAYER_TYPE].split(".")[-1]:
```

Comment:
@arvindsrikantan Add a comment on the check being made here. I find it very hard to understand how you conclude "incompatible" here.